addons/website_sale/static/src/js/dashboard/dashboard.js

```diff
@@ -0,0 +1,115 @@
+import { Component, onWillStart, useState } from '@odoo/owl';
+import { useService, useBus } from '@web/core/utils/hooks';
+import { DateFilterButton, DATE_OPTIONS } from '../date_filter_button/date_filter_button';
+
+export const CARD_COLORS_MAPPING = {
+    'to_fulfill': 'purple',
+    'to_confirm': 'orange',
+    'to_invoice': 'cyan',
+};
+
+export const CARD_FILTERS_MAPPING = {
+    'to_fulfill': ['to_fulfill', 'from_website', 'order_confirmed'],
+    'to_confirm': ['to_confirm', 'from_website'],
+    'to_invoice': ['to_invoice', 'from_website', 'order_confirmed'],
+};
+
+export const DEFAULT_FILTERS = ['from_website', 'order_confirmed'];
+
+export class Dashboard extends Component {
+	static template = 'website_sale.Dashboard';
+    static props = {};
+	static components = { DateFilterButton };
+
+	setup() {
+		this.state = useState({
+			eCommerceData: {},
+			selectedDateFilter: DATE_OPTIONS[0],
+			selectedCard: '',
+		});
+		this.orm = useService('orm');
+
+		useBus(this.env.searchModel, 'update', () => {
+            for (const [cardName, filters] of Object.entries(CARD_FILTERS_MAPPING)) {
+                if (this.isSameFilter(filters)) {
+                    return this.state.selectedCard = cardName;
+                }
+            }
+            this.state.selectedCard = '';
+		});
+
+		onWillStart(async () => {
+			await this.updateDashboardState();
+		});
+	}
+
+	async updateDashboardState(filter = false) {
+		if (filter) {
+			this.state.selectedDateFilter = filter;
+		}
+		this.state.eCommerceData = await this.orm.call(
+            'sale.order',
+            'retrieve_ecommerce_dashboard',
+            [this.state.selectedDateFilter.periodDays],
+        );
+	}
+
+	handleCardClick(ev) {
+        const cardName = ev.currentTarget.getAttribute('card_name');
+
+		if (this.state.selectedCard === cardName) {  //clicked already selected card
+			this.setFilters(DEFAULT_FILTERS);
+		} else {  //clicked another card
+			const filters = CARD_FILTERS_MAPPING[cardName];
+            this.setFilters(filters);
+		}
+	}
+
+	/**
+	 * This method clears the current search query and activates
+	 * the filters found in `filters`.
+	 */
+    setFilters(filters) {
+        const searchItems = this.env.searchModel.getSearchItems((item) =>
+			filters.includes(item.name)
+		);
+		this.env.searchModel.query = [];
+		for (const item of searchItems) {
+			this.env.searchModel.toggleSearchItem(item.id);
+		}
+    }
+
+	isSameFilter(filters) {
+		if (!filters) {
+			return false;
+		}
+		const activeSearchFilterNames = this.env.searchModel
+			.getSearchItems((el) => el.isActive && el.type === 'filter')
+			.map((el) => el.name);
+
+        if (filters.length !== activeSearchFilterNames.length) return false;
+
+        return filters.sort().every((val, i) => val === activeSearchFilterNames.sort()[i]);
```

Comment:
We could achieve same without using sort and sort in loop is even more bad (every is kind of loop)